https://bugs.gentoo.org/939721
https://github.com/ionenwks/iwdevtools/commit/4f3c463af0af4f62317f2117580e57beed3435b0
From: Ionen Wolkens <ionen@gentoo.org>
Date: Mon, 16 Sep 2024 02:48:31 -0400
Subject: [PATCH] tests: work around portage confusion by passing dev-test/

In theory portage should be ignoring ::gentoo (different repos.conf
which does not specify /var/db/repos/gentoo and changes main-repo),
but seems it gets confused by a package named "test" existing still
resulting in:

    You specified an unqualified atom that matched multiple packages:
    * dev-test/test
    * dev-python/test

Whether category is passed or not is not important in env-atom and
*-perms (so can just do it), but in "qa-cmp/pick" it was omitted on
a few tests to test the missing category logic.

So we lose that extra bit until can figure out a way for portage to
actually ignore ::gentoo. Not passing the category is however rarely
done given the portage bashrc hook passes it, this is for when using
qa-cmp directly and being brief with arguments.
--- a/tests/eoldnew/env-atom
+++ b/tests/eoldnew/env-atom
@@ -8,3 +8,3 @@ export EOLDNEW_EMERGE_ARGS_NEW="--new {}"
 export EOLDNEW_EMERGE_CMD=:
-expect -s "*'running: : =dev-test/test-1.5 --all test --more test --old test
-running: : test --all test --more test --new test'*" test
+expect -s "*'running: : =dev-test/test-1.5 --all dev-test/test --more dev-test/test --old dev-test/test
+running: : dev-test/test --all dev-test/test --more dev-test/test --new dev-test/test'*" dev-test/test
--- a/tests/qa-cmp/diff-filelist-perms
+++ b/tests/qa-cmp/diff-filelist-perms
@@ -9,6 +9,6 @@ expect -s "*'
  FILES:+usr/bin/file2 (-rw-r--r-- '*:*')
-'*'FILES(+1,-2)'*" -c test
+'*'FILES(+1,-2)'*" -c dev-test/test
 expect -s "*'
  FILES:-usr/bin/file3
-'*'FILES(-1)'*" -c --ignore-perms test
+'*'FILES(-1)'*" -c --ignore-perms dev-test/test
 expect -s "*'
@@ -17,3 +17,3 @@ expect -s "*'
  FILES:+usr/bin/file2 (-rw-r--r-- '*:*')
-'*'FILES(+1,-2)'*" -c --show-perms test
+'*'FILES(+1,-2)'*" -c --show-perms dev-test/test
 
@@ -23,2 +23,2 @@ expect -s "*'
  FILES:-usr/bin/file3
-'*'FILES(-1)'*" -c test
+'*'FILES(-1)'*" -c dev-test/test
--- a/tests/qa-cmp/pick
+++ b/tests/qa-cmp/pick
@@ -12,7 +12,10 @@ expect "*test-3.5*test-4.5*" -cW dev-test/test
 expect "*test-1.5*test-3.5*" -cW dev-test/test:2
-expect "*test-4.5*test-3.5*" -cW test =test-3.5
+expect "*test-4.5*test-3.5*" -cW dev-test/test =dev-test/test-3.5 # *1
 expect "*test-1.5*test-3.5*" -cW dev-test/test:2 =dev-test/test-3.5
-expect "*test-2.5*test-4.5*" -cW test:0 =test-4.5
-expect "*alt-1.5/image*test-4.5*" -cW alt test # nonsense but allowed
-expect "*=dev-test/alt-1.5*alt-1.5/image*" -cW alt
+expect "*test-2.5*test-4.5*" -cW dev-test/test:0 =dev-test/test-4.5 # *1
+expect "*alt-1.5/image*test-4.5*" -cW dev-test/alt dev-test/test # nonsense but allowed, *1
+expect "*=dev-test/alt-1.5*alt-1.5/image*" -cW dev-test/alt # *1
+# *1: meant to have these tests with unspecified category to test the category
+#     logic, but portage gets confused by ::gentoo (https://bugs.gentoo.org/939721)
+
 # FIXME:
--- a/tests/qa-cmp/single-perms
+++ b/tests/qa-cmp/single-perms
@@ -6,3 +6,3 @@ expect -s "*'
  FILES: usr/bin/file1 (-rwxr-xr-x '*:*')
- FILES: usr/bin/file2 (-rw-r--r-- '*:*')'*" -c --single-all --show-perms test
+ FILES: usr/bin/file2 (-rw-r--r-- '*:*')'*" -c --single-all --show-perms dev-test/test
 
@@ -11,2 +11,2 @@ expect -s "*'
  FILES: usr/bin/file1 (-rwxr-xr-x '*:*')
- FILES: usr/bin/file2 (?)'*" -c --single-all --show-perms test
+ FILES: usr/bin/file2 (?)'*" -c --single-all --show-perms dev-test/test
